pkg/cloudprovider/providers/openstack/openstack_loadbalancer.go

```diff
@@ -537,8 +551,8 @@ func (lbaas *LbaasV2) UpdateLoadBalancer(clusterName string, service *api.Servic
 
 	// Compose Set of member (addresses) that _should_ exist
 	addrs := map[string]empty{}
-	for _, nodeName := range nodeNames {
-		addr, err := getAddressByName(lbaas.compute, types.NodeName(nodeName))
+	for _, node := range nodes {
+		addr, err := nodeAddressForLB(node)
```

Comment:
The PR I had replaced `hostsFromNodeList` (which checks for unschedulable) with `getNodeConditionPredicate` (which checks for unschedulable and ready).  I needed to rewrite/remove `hostsFromNodeList` (because it uses node names) and I felt this was a correct change (in particular it makes `createLoadBalancer` and `nodeSyncLoop` use the same logic) - but you're right, I shouldn't have included that change in logic in this PR.

I've reverted this portion of the code to use the original logic (unschedulable only).